### PR TITLE
fix(rendering): allow render instructions to propagate while rendering slots

### LINKED ISSUE #9996: Fragment + nested client:only components + slots problem
### Astro Info

```block
Astro                    v4.3.2
Node                     v18.18.0
System                   macOS (arm64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/tailwind
                         @astrojs/react
                         simple-stream
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

below is the "not working" case, which is webpage display nothing.
```astro
---
import { Hello, Island, World } from "@components/demo.tsx";
---

<Island client:only>
  <Fragment slot="footer"><Hello client:only /></Fragment>
  <Fragment slot="header"><World client:only /></Fragment>
</Island>
```

but, if don't use Fragment, it works
```astro
<Island client:only>
  <a slot="footer"><Hello client:only /></a>
  <a slot="header"><World client:only /></a>
</Island>
```

also, below is work if nested components not defined `client:only`
```astro
<Island client:only>
  <Fragment slot="footer"><Hello /></Fragment>
  <Fragment slot="header"><World /></Fragment>
</Island>
```

### What's the expected result?

I expected this should work
```
<Island client:only>
  <Fragment slot="footer"><Hello client:only /></Fragment>
  <Fragment slot="header"><World client:only /></Fragment>
</Island>
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-qczw9r-srdto2?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes

- Fixes #9996
- The slot rendered first eagerly and stringified the island script with it. The slot was then put inside a template tag because the parent component is `client:only`.
- Island script should have been rendered right before the parent component itself. The `RenderDestination`-`RenderInstance` system makes sure that while things execute out of sequence, they get "stringified" in sequence.
- The reason was that slots are rendered in a temporary render destination and some chunk types are not handled there, falling back to stringifying immediately.
- The fix is to defer stringifying to the real destination, accumulating instructions in the `SlotString` untouched.

## Testing
- Added a case in an existing E2E fixture.

## Docs
- Does not affect usage.

### CODE REVIEW

File: packages/astro/playwright.config.js [hunk 1]
```diff
@@ -27,16 +27,6 @@ const config = {
...
-	projects: [
-		{
-			name: 'Chrome Stable',
-			use: {
-				browserName: 'chromium',
-				channel: 'chrome',
-				args: ['--use-gl=egl'],
-			},
-		},
-	],
```
Comment (ematipico) [inline packages/astro/playwright.config.js:39]: Why was this removed?

**Review by ematipico — APPROVED** [review]
I would ask @bluwy and @matthewp to look at the PR. It seems good to me though

**Review by matthewp — APPROVED** [review]

File: packages/astro/playwright.config.js (hunk 1 above)
Comment (lilnasy) [reply packages/astro/playwright.config.js:39]: I need this to run E2E tests locally. It shouldn't have been committed. I'll undo it.

**Review by bluwy — APPROVED** [review]
